### PR TITLE
Update siglensent helm charts

### PR DESCRIPTION
**Pull request by venkateshamatam**
# Description 

Steps to deploy these Helm charts: 

1. Clone the repo and `cd charts`
2. Make sure to update `charts/siglensent/values.yaml` with the appropriate license key and host names. If you do not have a host name, you can use a self-signed certificate. 
3. Run `helm install siglensent ./siglensent -n siglensent -f ./siglensent/Values.yaml`
4. Verify that the services and pods are created:
  `kubectl get services -n siglensent` 
  `kubectl get pods -n siglensent`

### CODE REVIEW

File: charts/siglensent/templates/raft-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ssInstanceName: {{ .Values.config.raft.ssInstanceName }}
+    rbac:
+      provider: sqlite
+      dbname: db1
+    ingestNode: true
+    queryNode: true
+    etcd:
+      enabled: true
+      username: root
+      password: dontkeepthispassword
```
Comment (AndrewHess) [inline charts/siglensent/templates/raft-configmap.yaml:26]: 1. The username and password should be in values.yaml so it's easier for users to change (and because it's duplicated in the raft-configmap and the worker-configmap)
2. The `enabled` should also be in values.yaml, unless you have some other config in values.yaml that will make it so this raft-configmap.yaml doesn't do anything when the chart is used in a single-node configuration (since in that configuration, we don't need raft)

File: charts/siglensent/templates/raft-configmap.yaml
Comment (AndrewHess) [inline charts/siglensent/templates/raft-configmap.yaml:1]: I'm not seeing `isRaftNode: true` anywhere. Am I just missing it, or do you not need it?

File: charts/siglensent/templates/ingest-service.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: ingest-service
+  namespace: {{ .Values.global.namespace }}
+spec:
+  selector:
+    app: {{ .Values.ingestService.selector.app }}
+  ports:
+    - protocol: TCP
+      port: 8081
+      targetPort: 8081
```
Comment (AndrewHess) [inline charts/siglensent/templates/ingest-service.yaml:12]: These should be configurable in values.yaml

Similar comment for query-service.yaml

File: charts/siglensent/templates/ingress.yaml [hunk 3]
```diff
@@ -22,7 +22,7 @@ spec:
         pathType: Prefix
         backend:
           service:
-            name: {{ .Release.Name }}-query-service
+            name: query-service
```
Comment (AndrewHess) [inline charts/siglensent/templates/ingress.yaml:25]: Why don't you want this release name prefix? (It's been a while since I've used helm, so I might be forgetting something)

File: charts/siglensent/templates/worker-configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    ingestNode: true
+    queryNode: true
+    etcd:
+      enabled: true
+      username: root
+      password: dontkeepthispassword
+      seedUrls:
+        - {{ printf "http://%s-etcd.%s.svc.cluster.local:2379" .Release.Name .namespace }}
+    raft:
+      enabled: false
```
Comment (AndrewHess) [inline charts/siglensent/templates/worker-configmap.yaml:30]: This doesn't seem right. Does this work for a multinode cluster (e.g., sending a query gets all the data from the cluster, not just the data from one worker)? I expected something like:
```
isRaftNode: false
raft:
    enabled: true
    username: ...
    password: ...
    raftNodes: ...
```

File: charts/siglensent/values.yaml [hunk 5]
```diff
@@ -8,55 +8,36 @@ ingress-nginx:
...
-dockerConfigBase64: ""
-
-image: 123456789012.dkr.ecr.us-east-1.amazonaws.com/siglens:latest
-imagePullPolicy: Always
-
-# If you're using an enterprise image, set "enterprise" to true. Enterprise
-# images require a license. To specify your license, set "licenseBase64" to the
-# output of `cat /path/to/licenseKey.txt | base64`.
-enterprise: false
-licenseBase64: ""
```
Comment (AndrewHess) [inline charts/siglensent/values.yaml:31]: You replaced this with
```
secrets:
  licenseKey: ""
```
Should people using this chart set this to the raw text in the license file? Or to something else (e.g., the path to the file)? The license file has binary data, so there might be some issues with directly copy-pasting that here, which is why we were doing a base64 encoding previously. Did you not have issues with your new way?

File: charts/siglensent/values.yaml [hunk 6]
```diff
@@ -8,55 +8,36 @@ ingress-nginx:
...
-  namespace: siglensent
-  replicaCount: 3
-  auth:
-    rbac:
-      rootPassword: dontkeepthispassword
-
-# Used for pulling the SigLens image from a private registry. Not needed for
-# public registries. If you're using a private registry, set this to the output
-# of `cat ~/.docker/config.json | base64`.
-dockerConfigBase64: ""
```
Comment (AndrewHess) [inline charts/siglensent/values.yaml:22]: You moved this to:
```
registry:
  dockerConfigBase64: ""
```
And that's fine, but why did you remove the comment? I think the comment will help clarify exactly what people using the chart should set the value to

**Review by AndrewHess — COMMENTED** [review]
One note: there's quite a few values (ports, storage size, etc.) that you've hard coded inside a template file. These should all be in values.yaml and then those values should be read into the template. Basically, the templates files should give the basic structure of how we deploy to k8s, but if someone deploying the chart wants to make any change about specifics they should only need to modify values.yaml

File: charts/siglensent/templates/worker-configmap.yaml (hunk 4 above)
Comment (venkateshamatam) [reply charts/siglensent/templates/worker-configmap.yaml:30]: Yes it worked without `isRaftNode` set. I ingested different data into each worker node and running a query gets all data from the cluster.

File: charts/siglensent/values.yaml (hunk 5 above)
Comment (venkateshamatam) [reply charts/siglensent/values.yaml:31]: The new licenseKey should also be base64 encoded, I added the comment back. It's the same but just renamed from `licenseBase64` to "secrets.licenseKey" in `Values.yaml`

**Comment by venkateshamatam** [comment]
@AndrewHess  I addressed all comments, but with `isRaftNode: true` the worker-deployment pods fail to start with this error. There is no issue with raft-deployment pods.

```
time="2024-08-27 14:26:38" level=error msg="etcd.Init: Failed to get member list from etcd: context deadline exceeded"
time="2024-08-27 14:26:38" level=error msg="InitalizeDistributedStorage: Failed to initialize etcd storage: context deadline exceeded"
time="2024-08-27 14:26:38" level=error msg="Failed to init distributed storage: context deadline exceeded"
time="2024-08-27 14:26:38" level=warning msg="Flushing 0 segment files on server exit"
{"level":"warn","ts":"2024-08-27T14:26:38.124686Z","logger":"etcd-client","caller":"v3@v3.5.13/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc0001b7680/raft-deployment-0.raft-service.siglensent.svc.cluster.local:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing: dial tcp 10.24.0.24:2379: connect: connection refused\""}
time="2024-08-27 14:26:38" level=warning msg="FlushAliasMapToFile: Flushing alias map to file on exit"
time="2024-08-27 14:26:38" level=info msg="ForcedFlushToScrollFile: Flushing scroll context to file"
time="2024-08-27 14:26:38" level=error msg="siglens main: Error in starting server:context deadline exceeded "
```

**Comment by AndrewHess** [comment]
> @AndrewHess I addressed all comments, but with `isRaftNode: true` the worker-deployment pods fail to start with this error. There is no issue with raft-deployment pods. Could you confirm if the config is right or if something is missing?

The worker config looks correct now. Also I believe the reason you're only seeing this issue now is because now in the `raft` section you have `enabled: true` (which is correct); the issue isn't that you have `isRaftNode: false` (which is also correct). Before, when you had `enabled: false`, the workers weren't trying to connect to the raft cluster

File: charts/siglensent/templates/raft-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+      containers:
+      - name: raft-container
+        image: {{ .Values.raft.deployment.image }}
+        imagePullPolicy: {{ .Values.raft.deployment.imagePullPolicy }}
+        command: ["/bin/sh"]
+        args:
+          - -c
+          - |
+            sleep 10
+            exec /siglens/sigscalr --config /config/server.yaml
```
Comment (AndrewHess) [inline charts/siglensent/templates/raft-deployment.yaml:49]: Why is `sleep 10` necessary?

Comment (AndrewHess) [reply charts/siglensent/templates/raft-deployment.yaml:49]: nit: in worker-deployment.yaml you start like this:
```
        command: ["/siglens/sigscalr"]
        args: ["--config", "/config/server.yaml"]
```
Why don't you start them both the same?

File: charts/siglensent/Chart.yaml [hunk 8]
```diff
@@ -25,8 +25,4 @@ dependencies:
     repository: https://kubernetes.github.io/ingress-nginx
   - name: cert-manager
     version: "1.14.4"
-    repository: https://charts.jetstack.io
-  - name: etcd
-    version: "10.0.3"
-    repository: https://charts.bitnami.com/bitnami
-    condition: multinode.enabled
+    repository: https://charts.jetstack.io
```
Comment (AndrewHess) [inline charts/siglensent/Chart.yaml:28]: Removing the `etcd` chart is going to break multinode configured with S3. I'm pretty sure some other changes in this PR will also break S3 multinode.

@nkunal do we want to accept this PR with these breaking changes, or take a little more time to ensure we don't break the S3 configuration?

File: charts/siglensent/values.yaml [hunk 9]
```diff
@@ -93,5 +88,33 @@ config:
...
+    storage:
+      size: "10Gi"
+
+worker:
+  deployment:
+    replicas: 3 # Number of replicas for the Worker server
+    image: public.ecr.aws/l3t9n5v5/siglens-enterprise:0.2.21-ent1
+    imagePullPolicy: Always
+    ports:
+      ingest: 8081
```
Comment (AndrewHess) [inline charts/siglensent/values.yaml:113]: nit: I can't think of a reason someone would want a different image or pull policy for the Raft and Worker nodes. I think you should have those outside the `raft:` and `worker:` sections (maybe in the `registry:` section?), unless there's a specific reason you configure them twice.

Comment (AndrewHess) [reply charts/siglensent/values.yaml:113]: nit: why does the `raft` section have `ingest: 8081`? Why does it need to know which port the workers use for ingest?

File: charts/siglensent/values.yaml [hunk 10]
```diff
@@ -8,55 +8,50 @@ ingress-nginx:
...
 
-queryHost: "siglens.example.com"
-ingestHost: "ingest.siglens.example.com"
-
-ebs:  # Used for sharing data between pods on the same node.
-  storageSize: 20Gi
+# images require a license. To specify your license, set "licenseBase64" to the
+# output of `cat /path/to/licenseKey.txt | base64`.
+secrets:
+  licenseKey: ""
```
Comment (AndrewHess) [inline charts/siglensent/values.yaml:51]: nit: you should probably include `base64` in the `licenseKey` name, so it will be more clear. It used to be called `licenseBase64`

File: charts/siglensent/templates/raft-configmap.yaml [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    dataPath: {{ .Values.config.dataPath }}
+    timestampKey: {{ .Values.config.timestampKey }}
+    licenseKeyPath: {{ .Values.config.licenseKeyPath }} 
+    ssInstanceName: {{ .Values.config.raft.ssInstanceName }}
+    rbac:
+      provider: sqlite
+      dbname: db1
+    ingestNode: true
+    queryNode: true
+    isRaftNode: true
```
Comment (AndrewHess) [inline charts/siglensent/templates/raft-configmap.yaml:23]: For raft nodes, you should have:
```
ingestNode: false
queryNode: false
```
(but keep `isRaftNode: true`)

File: charts/siglensent/templates/raft-deployment.yaml [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+        ports:
+        - containerPort: {{ .Values.raft.deployment.ports.ingest }}
+        - containerPort: {{ .Values.raft.deployment.ports.raft }}  # Raft port
```
Comment (AndrewHess) [inline charts/siglensent/templates/raft-deployment.yaml:52]: nits:
1. Why do you need the `ingest` port?
2. ` # Raft port` is already clear from the code, so you can delete this comment

File: charts/siglensent/templates/raft-deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+          name: raft-configmap
+      - name: license-volume
+        secret:
+          secretName: {{ .Release.Name }}-license-secret
+          items:
+          - key: license
+            path: license.txt
+      - name: cert-volume
+        secret:
+          secretName: ingress-tls
```
Comment (AndrewHess) [inline charts/siglensent/templates/raft-deployment.yaml:82]: Somewhere else you looked up `ingress-tls` from values.yaml. Either look it up everywhere, or hardcode it everywhere.

I think a user of this chart would never have a reason to change `ingress-tls` to something else, is that right? If it is, I don't see a problem with hardcoding it. If it's not, you definitely want to have it configurable in values.yaml

File: charts/siglensent/templates/raft-service.yaml [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: raft-service
+  namespace: {{ .Values.global.namespace }}
+spec:
+  clusterIP: None
+  selector:
+    app: raft-node
+  ports:
+    - port: 2379
+      targetPort: 2379
```
Comment (AndrewHess) [inline charts/siglensent/templates/raft-service.yaml:12]: Don't you need to read these from values.yaml?

File: charts/siglensent/templates/worker-deployment.yaml [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+          name: worker-configmap
+      - name: license-volume
+        secret:
+          secretName: {{ .Release.Name }}-license-secret
+          items:
+          - key: license
+            path: license.txt
+      - name: cert-volume
+        secret:
+          secretName: ingress-tls
```
Comment (AndrewHess) [inline charts/siglensent/templates/worker-deployment.yaml:58]: Here you also hardcode `ingress-tls`

File: charts/siglensent/values.yaml [hunk 16]
```diff
@@ -8,55 +8,50 @@ ingress-nginx:
...
-dockerConfigBase64: ""
+pvc:
+  size: 20Gi
 
-image: 123456789012.dkr.ecr.us-east-1.amazonaws.com/siglens:latest
-imagePullPolicy: Always
+etcd:
+  enabled: true
+  username: root
+  password: dontkeepthispassword
```
Comment (AndrewHess) [inline charts/siglensent/values.yaml:20]: Why did you change this `etcd` config? It used to be:
```
# Overrides for the etcd subchart.
etcd:
  namespace: siglensent
  replicaCount: 3
  auth:
    rbac:
      rootPassword: dontkeepthispassword
```
And I think it's only used by the `etcd` chart which is used for multinode in S3 configuration. But you removed that chart in this PR

File: charts/siglensent/values.yaml [hunk 17]
```diff
@@ -8,55 +8,50 @@ ingress-nginx:
...
+  selector:
+    app: worker-node
+  port: 5122
+  targetPort: 5122
+
+ingress:
+  hosts:
+    - siglens.example.com
+    - ingest.siglens.example.com
+  tlsSecret: ingress-tls
```
Comment (AndrewHess) [inline charts/siglensent/values.yaml:41]: You have some duplication. These URLs are what's specified a ways above:
```
queryHost: siglens.example.com
ingestHost: ingest.siglens.example.com
```
Can we remove this duplication? If not, it would be good to at least move these two sections close to each other so that if a user of this chart has to modify them, they'll be less likely to accidentally modify one section and forget to modify the other section

File: charts/siglensent/values.yaml [hunk 18]
```diff
@@ -93,5 +88,30 @@ config:
...
+      ingest: 8081
+
+
+# Used for pulling the SigLens image from a private registry. Not needed for
+# public registries. If you're using a private registry, set this to the output
+# of `cat ~/.docker/config.json | base64`.
+registry:
+  dockerConfigBase64: ""
+  image: public.ecr.aws/l3t9n5v5/siglens-enterprise:0.2.21-ent1
+  imagePullPolicy: Always
```
Comment (AndrewHess) [inline charts/siglensent/values.yaml:117]: nit: clarify that this comment is referring only to `dockerConfigBase64`. And maybe move the comment inside the `registry` block?

**Review by AndrewHess — APPROVED** [review]
